core/storage/base_model/gae_models.py

```diff
@@ -33,54 +55,60 @@
 # method to find the location of this delimiter.
 VERSION_DELIMITER = '-'
 
+# Constant used when retrieving big number of models.
+FETCH_BATCH_SIZE = 1000
+
+# Constants used for generating ids.
+MAX_RETRIES = 10
+RAND_RANGE = (1 << 30) - 1
+ID_LENGTH = 12
+
+
 # Types of deletion policies. The pragma comment is needed because Enums are
 # evaluated as classes in Python and they should use PascalCase, but using
 # UPPER_CASE seems more appropriate here.
-DELETION_POLICY = python_utils.create_enum(  # pylint: disable=invalid-name
+class DELETION_POLICY(enum.Enum): # pylint: disable=invalid-name
```

Comment:
What is the effect of this change on existing models in the datastore? /cc @vojtechjelinek

Reply:
(Note: @hardikkat24, to be clear, this is a question for you. You can test by making some models in develop on your local dev server and then switching to this branch. I'm only cc-ing @vojtechjelinek for info.)

Reply:
I had a look and talked to Vojta about this.
Here we don't actually store our ENUMs in the datastore. We only return them from methods like: `get_deletion_policy()`, `get_model_association_to_user()`, `get_export_policy()`.
Still I changed our ENUMs from int to str, like `KEEP = 1` to `KEEP = 'KEEP'` to keep our ENUMs similar to enums we had.

Reply:
Though not stored, but ENUMS are part of the procedure how models are managed in datastore, and I assume that was the focus point here. But with tests like [these](https://github.com/oppia/oppia/blob/d2831581de5b7665a59599ead5512ef076df5c5e/core/storage/user/gae_models_test.py#L114), wouldn't it be automatically tested whether new implementation is working fine or not (i.e. managing model creation/deletion/export etc the same old way or not).

Reply:
Ah actually, on further investigation, these are totally fine either way, because we don't store the enums in the datastore as @hardikkat24 mentioned. So I'm no longer worried and would be OK with anything as the values (either numbers or strings).

For any enums that are used as choices for something to store in the datastore, though, we should be careful. Otherwise a change to our representation here can mess up the old models which would become invalid since the field value they store is no longer part of the enum.